src/Estimators/src/BaseEstimatorFromFootIMU.cpp

```diff
@@ -0,0 +1,263 @@
+
+    if (!m_isModelSet)
+    {
+        log()->error("{} No iDynTree::Model has been set.", logPrefix);
+        return false;
+    }
+
+    if (!m_isStanceFootSet)
+    {
+        log()->error("{} No StanceFootRF has been set.", logPrefix);
+        return false;
+    }
+
+    if (!m_kinDyn.isValid())
+    {
+        log()->error("{} iDynTree::KinDynComputations m_kinDyn is not valid.", logPrefix);
+        return false;
+    }
+
+    m_gravity << 0, 0, -9.81;  // TO DO: verify the reference frame
+    m_frameIndex = m_kinDyn.getFrameIndex(BaseEstimatorFromFootIMU::STANCE_FOOT_TO_STRING(m_stanceFootRF));
+    m_frameName = m_kinDyn.getFrameName(m_frameIndex);
+    m_linkIndex = m_model.getFrameLink(m_frameIndex);
+
+    if (!m_kinDyn.setFloatingBase(m_model.getLinkName(m_linkIndex)))
+    {
+        log()->error("{} Unable to set the floating base.", logPrefix);
+        return false;
+    }
+
+    iDynTree::Transform frame_H_link = m_model.getFrameTransform(m_frameIndex).inverse();
+    m_frame_H_link = BipedalLocomotion::Conversions::toManifPose(frame_H_link);
+
+    auto ptr = handler.lock();
+    if (ptr == nullptr)
+    {
+        log()->error("{} Invalid parameter handler.", logPrefix);
+        return false;
+    }
+
+    auto populateparameter
+        = [logPrefix](std::shared_ptr<const ParametersHandler::IParametersHandler> ptr,
+        const std::string& paramName, auto& parameter) -> bool {
+        if (!ptr->getParameter(paramName, parameter))
+        {
+            log()->error("{} Unable to find the parameter '{}'", logPrefix, paramName);
+            return false;
+        }
+        return true;
+    };
+
+    //Params params;
+    bool ok = populateparameter(ptr->getGroup("MODEL_INFO").lock(), "base_frame", m_baseFrameName);     //root link of the robot (whose pose must be estimated)
+    ok = ok && populateparameter(ptr, "foot_width_in_m", m_footWidth);
+    ok = ok && populateparameter(ptr, "foot_length_in_m", m_footLength);
+    //ok = ok && populateparameter("switch_on_after", params.switchOnAfter);
+
+    m_isInitialized = true;
+
+    return ok;
+}
+
+bool BaseEstimatorFromFootIMU::setInput(const BaseEstimatorFromFootIMUInput& input)
+{
+    m_input = input;                    //TO DO: check if this line goes wrong
+    return true;
+}
+
+bool BaseEstimatorFromFootIMU::advance()
+{
+    constexpr auto logPrefix = "[BaseEstimatorFromFootIMU::advance]";
+
+    m_isOutputValid = false;
+
+    if (!m_isInitialized)
+    {
+        log()->error("{} The estimator is not initialized.", logPrefix);
+        return false;
+    }
+
+    // Set the 4 foot vertices in World reference frame [dimensions in meters]
+    m_p1 << +m_footWidth / 2, +m_footLength / 2, 0;
+    m_p2 << -m_footWidth / 2, +m_footLength / 2, 0;
+    m_p3 << -m_footWidth / 2, -m_footLength / 2, 0;
+    m_p4 << +m_footWidth / 2, -m_footLength / 2, 0;
+
+    Eigen::Vector3d desiredTranslation = m_input.desiredFootPose.translation();
```

Comment:
can you clean this part of the code and comments that better explain the passages?

Reply:
ok done.